src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -657,6 +661,39 @@ public function handleBlockActorData(BlockActorDataPacket $packet) : bool{
 			}
 
 			$this->session->getLogger()->debug("Invalid sign update data: " . base64_encode($packet->nbt->getEncodedNbt()));
+		}elseif($block instanceof Beacon){
+			$inventory = $this->player->getCurrentWindow();
+			if(!($inventory instanceof BeaconInventory)){
+				return false;
+			}
+			$itemId = $inventory->getInput()->getId();
+			if(!isset(Beacon::ALLOWED_ITEM_IDS[$itemId])){
+				throw new PacketHandlingException("Invalid input $itemId");
+			}
+
+			if(($beaconLevel = $block->getBeaconLevel()) > 0 && $block->viewSky()){
+				$allowedEffects = $block->getAllowedEffect($block->getBeaconLevel());
+				$primaryEffect = EffectIdMap::getInstance()->fromId($nbt->getInt("primary", 0));
+				if($primaryEffect !== null && in_array($primaryEffect, $allowedEffects, true)){
+					$event = new BeaconActivateEvent($block, $primaryEffect);
```

Comment:
Does we really need a event?

Reply:
I think the main reason why not everything has an event right now is that an event file has too much boilerplate (create a new file, add license headers, generate getters/setters). Looks fine to me if it is already created.

Reply:
I think that an event, in this case, is interesting, especially for the plugins that would like to modify the actions done.